prov/efa/src/efa_rma.c

```diff
@@ -225,7 +227,12 @@ static inline ssize_t efa_rma_post_write(struct efa_base_ep *base_ep,
 		ibv_wr_start(qp->ibv_qp_ex);
 		base_ep->is_wr_started = true;
 	}
-	qp->ibv_qp_ex->wr_id = (uintptr_t) ((flags & FI_COMPLETION) ? msg->context : NULL);
+
+	qp->ibv_qp_ex->wr_id = (uintptr_t) efa_fill_context(
+		msg->context, msg->addr, flags,
+		flags & FI_REMOTE_CQ_DATA ?
+			FI_REMOTE_CQ_DATA | FI_RMA | FI_REMOTE_WRITE :
```

Comment:
This is not right, `FI_REMOTE_WRITE` for the target of the fi_writedata.

`FI_REMOTE_CQ_DATA` as a completion flag is also for the target of `fi_*data` operation, You shouldn't have `FI_REMOTE_CQ_DATA` in the initiator of a `data` operation.

For this change, you don't need any check on `flags` for your completion flags, it should just be `FI_RMA | FI_WRITE`